pkg/operator/configobservation/auth/webhook_authenticator.go

```diff
@@ -0,0 +1,212 @@
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+)
+
+var (
+	webhookTokenAuthenticatorPath = []string{"apiServerArguments", "authentication-token-webhook-config-file"}
+	webhookTokenAuthenticatorFile = []interface{}{"/etc/kubernetes/static-pod-resources/secrets/webhook-authenticator/kubeConfig"}
+)
+
+func ObserveWebhookTokenAuthenticator(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, _ []error) {
+	defer func() {
+		configobserver.Pruned(ret, webhookTokenAuthenticatorPath)
+	}()
+
+	listers := genericListers.(configobservation.Listers)
+	resourceSyncer := genericListers.ResourceSyncer()
+
+	errs := []error{}
+	existingWebhookAuthenticator, _, err := unstructured.NestedSlice(existingConfig, webhookTokenAuthenticatorPath...)
+	if err != nil {
+		errs = append(errs, err)
+	}
+	existingWebhookConfigured := len(existingWebhookAuthenticator) > 0
+
+	observedConfig := map[string]interface{}{}
+
+	auth, err := listers.AuthConfigLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		return observedConfig, nil
+	} else if err != nil {
+		return existingConfig, []error{err}
+	}
+
+	var webhookSecretName string
+	if auth.Spec.WebhookTokenAuthenticator != nil {
+		webhookSecretName = auth.Spec.WebhookTokenAuthenticator.KubeConfig.Name
+	}
+
+	observedWebhookConfigured := len(webhookSecretName) > 0
+	if observedWebhookConfigured {
+		// retrieve the secret from config and validate it, don't proceed on failure
+		kubeconfigSecret, err := listers.ConfigSecretLister().Secrets("openshift-config").Get(webhookSecretName)
+		if err != nil {
+			return existingConfig, []error{fmt.Errorf("secret openshift-config/%s: %w", webhookSecretName, err)}
+		}
+
+		if secretErrors := validateKubeconfigSecret(kubeconfigSecret); len(secretErrors) > 0 {
+			return existingConfig, append(errs,
+				fmt.Errorf("secret openshift-config/%s: %w", webhookSecretName, utilerrors.NewAggregate(secretErrors)))
+		}
+
+		if err := unstructured.SetNestedField(observedConfig, webhookTokenAuthenticatorFile, webhookTokenAuthenticatorPath...); err != nil {
+			return existingConfig, append(errs, err)
+		}
+
+		resourceSyncer.SyncSecret(
+			resourcesynccontroller.ResourceLocation{Namespace: operatorclient.TargetNamespace, Name: "webhook-authenticator"},
+			resourcesynccontroller.ResourceLocation{Namespace: operatorclient.GlobalUserSpecifiedConfigNamespace, Name: webhookSecretName},
+		)
```

Comment:
syncing from an observer is an unexpected side-effect. How do we do this elsewhere?

Reply:
This is being done in other observers as well:
https://github.com/openshift/cluster-kube-apiserver-operator/blob/f73bebb6361c3649dab5305d8c7d1cd9753e61aa/pkg/operator/configobservation/apiserver/observe_apiserver.go#L226
https://github.com/openshift/cluster-kube-apiserver-operator/blob/f73bebb6361c3649dab5305d8c7d1cd9753e61aa/pkg/operator/configobservation/auth/auth_metadata.go#L84-L93
We're handling the field that's a reference to a secret here, it makes sense to me to handle the sync as well.

Reply:
This is backed by the resource sync controller, which then essential does this:
```
	c.configMapSyncRules[destination] = source
```

So this sync is ongoing, the call only registers a rule, does not do the a synchronous sync. And a new, different source overrides the old source.

So, I think this is fine.